### ISSUE TITLE
Exception on `ORGANIZER` field parsing

### ISSUE THREAD

**Issue by Toshik** [TesteurManiak/icalendar_parser#33]
There is an issue with parsing ICS containing partial `ORGANIZER` field (with no email specified):

```
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
BEGIN:VEVENT
UID:ac6d76d2-deb8-421d-86c6-670c06d5cd8c
SUMMARY:Cool event
DTSTAMP:20211004T074300Z
DTSTART:20211006T063000Z
DTEND:20211006T073000Z
DESCRIPTION:Some description
ORGANIZER;CN=Joe Jackson
END:VEVENT
END:VCALENDAR
```

Produces following  exception:
```
type 'String' is not a subtype of type 'List<String>' of 'other'
```
at 
```
#1      ICalendar.fromListToJson (package:icalendar_parser/src/model/icalendar.dart:253:35)
```

**Comment by TesteurManiak** [comment]
I've tested your file and execpt that it does not have a `PRODID` field (which is mandatory) I cannot seem to reproduce your issue. Please add some details to your issue such as which version of the package you are using and a minimal reproducible example (or even better a failing unit test).

The current output I'm getting with your ics file when testing it is the following:

```bash
> dart test .\test\parsing_test.dart
Building package executable...
Built test:test.
00:01 +11 -1: Register fields No email in ORGANIZER [E]                                                                                                                                                                                            
  ICalendarFormatException: The body is missing the property PRODID
  package:icalendar_parser/src/model/icalendar.dart 288:7  ICalendar.fromListToJson
  package:icalendar_parser/src/model/icalendar.dart 61:24  new ICalendar.fromLines
  test\parsing_test.dart 107:17                            main.<fn>.<fn>
```

And by adding a `PRODID` and re-testing I don't have any errors:

## Tested file (after adding PRODID)

```ics
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//hacksw/handcal//NONSGML v1.0//EN
CALSCALE:GREGORIAN
BEGIN:VEVENT
UID:ac6d76d2-deb8-421d-86c6-670c06d5cd8c
SUMMARY:Cool event
DTSTAMP:20211004T074300Z
DTSTART:20211006T063000Z
DTEND:20211006T073000Z
DESCRIPTION:Some description
ORGANIZER;CN=Joe Jackson
END:VEVENT
END:VCALENDAR
```

## Unit test

```dart
test('No email in ORGANIZER', () {
  final eventText = readFileLines('organizer_no_email.ics');
  ICalendar.fromLines(eventText);
});
```

## Output

```bash
> dart test .\test\parsing_test.dart
00:00 +13: All tests passed!
```

**Comment by Toshik** [comment]
My bad, I guess it is related to the `STATUS` and `CATEGORIES` fields. Maybe some others as well

- This one fails with 

```
NoSuchMethodError: Class 'IcsStatus' has no instance method '+'
```

```
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
PRODID:adamgibbons/ics
METHOD:PUBLISH
BEGIN:VEVENT
UID:b11ae923-9898-44ca-8f92-ed2ebe36863f
SUMMARY:Cool event
DTSTAMP:20211004T101600Z
DTSTART:20211006T063000Z
DTEND:20211006T073000Z
DESCRIPTION:Some description
ORGANIZER;CN=Joe Jackson
STATUS:CANCELLED
ORGANIZER;CN=Joe Jackson
END:VEVENT
END:VCALENDAR
```

- This one fails with
```
type 'String' is not a subtype of type 'List<String>' of 'other'
```
```
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
PRODID:adamgibbons/ics
METHOD:PUBLISH
BEGIN:VEVENT
UID:b11ae923-9898-44ca-8f92-ed2ebe36863f
SUMMARY:Cool event
DTSTAMP:20211004T101600Z
DTSTART:20211006T063000Z
DTEND:20211006T073000Z
DESCRIPTION:Some description
CATEGORIES:APPOINTMENT
ORGANIZER;CN=Joe Jackson
END:VEVENT
END:VCALENDAR
```

**Comment by TesteurManiak** [comment]
Your issue should be fixed in the newest release [1.0.1 of the package](https://pub.dev/packages/icalendar_parser/versions/1.0.1)
